source/crud/tests/unified/insertOne-dots_and_dollars.yml

```diff
@@ -0,0 +1,241 @@
+    expectEvents: &expectEventsDollarPrefixedKey
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dollarPrefixedKey
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dollarPrefixedKey
+
+  - description: "Inserting document with top-level dollar-prefixed key on pre-5.0 server yields server-side error"
+    runOnRequirements:
+      - maxServerVersion: "4.99"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: *dollarPrefixedKey
+        expectError:
+          isClientError: false
+    expectEvents: *expectEventsDollarPrefixedKey
+    outcome: *initialData
+
+  - description: "Inserting document with top-level dotted key"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: &dottedKey { _id: 1, a.b: 1 }
+        expectResult: *insertResult
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dottedKey
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dottedKey
+
+  - description: "Inserting document with dollar-prefixed key in embedded doc"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: &dollarPrefixedKeyInEmbedded { _id: 1, a: { $b: 1 } }
+        expectResult: *insertResult
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dollarPrefixedKeyInEmbedded
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dollarPrefixedKeyInEmbedded
+
+  - description: "Inserting document with dotted key in embedded doc"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: &dottedKeyInEmbedded { _id: 1, a: { b.c: 1 } }
+        expectResult: *insertResult
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dottedKeyInEmbedded
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dottedKeyInEmbedded
+
+  - description: "Inserting document with dollar-prefixed key in _id yields server-side error"
+    # Note: 5.0+ did not remove restrictions on dollar-prefixed keys in _id documents
```

Comment:
The scope originally expected this would work, but the functionality was removed in the final server implementation. The note here is just to clarify why we're not limiting this test to a particular server version.